Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -38,6 +38,23 @@ def relocate_dynamic_linkage(relocation)
     end
   end
 
+  # Ensures that an ad-hoc signature is present on all binaries,
+  # using the OS's builtin codesign tool.
+  # This is probably temporary - while codesigning is required on
+  # Apple Silicon, the build tools should be able to generate valid
+  # ad-hoc signatures.
+  # That does not fully work at this time, so Homebrew will build
+  # non-working software without this step.
+  def codesign_mach_o_files
+    return unless Hardware::CPU.arm? && MacOS.version == :big_sur
+
+    mach_o_files.each do |file|
+      file.ensure_writable do
+        system("codesign", "-s", "-", file)
```

Comment:
```suggestion
        system("codesign", "-s", "-", "--preserve-metadata=identifier,entitlements,flags,runtime", "-f", file)
```

- Without `--preserve-metadata=`, if the package itself does sign the binary purposely, though unlikely, information such as hardened runtime adoption and special entitlements requirement will get lost upon resigning.
- Without `-f`, `codesign` will fail if there is an existing signature, even when it is invalid.